mailet/api/src/main/java/org/apache/mailet/AttributeValue.java

```diff
@@ -0,0 +1,205 @@
+    public static AttributeValue<Collection<AttributeValue<?>>> of(Collection<AttributeValue<?>> value) {
+        return new AttributeValue<>(value, new Serializer.CollectionSerializer());
+    }
+
+    @SuppressWarnings({ "unchecked", "rawtypes" })
+    public static AttributeValue<Map<String, AttributeValue<?>>> of(Map<String, AttributeValue<?>> value) {
+        return new AttributeValue<>(value, new Serializer.MapSerializer());
+    }
+
+    public static AttributeValue<Serializable> ofSerializable(Serializable value) {
+        return new AttributeValue<>(value, new Serializer.FSTSerializer());
+    }
+
+    @SuppressWarnings("unchecked")
+    public static AttributeValue<?> ofAny(Object value) {
+        if (value instanceof Boolean) {
+            return of((Boolean) value);
+        }
+        if (value instanceof String) {
+            return of((String) value);
+        }
+        if (value instanceof Integer) {
+            return of((Integer) value);
+        }
+        if (value instanceof Long) {
+            return of((Long) value);
+        }
+        if (value instanceof Float) {
+            return of((Float) value);
+        }
+        if (value instanceof Double) {
+            return of((Double) value);
+        }
+        if (value instanceof Collection<?>) {
+            return of(((Collection<AttributeValue<?>>) value));
+        }
+        if (value instanceof Map<?,?>) {
+            return of(((Map<String, AttributeValue<?>>) value));
+        }
+        if (value instanceof QueueSerializable) {
+            return of((QueueSerializable) value);
+        }
+        if (value instanceof URL) {
+            return of((URL) value);
+        }
+        if (value instanceof Serializable) {
+            return ofSerializable((Serializable) value);
+        }
+        throw new IllegalArgumentException("input should at least be Serializable");
+    }
+
+    public static AttributeValue<?> fromJsonString(String json) throws IOException {
+        ObjectMapper objectMapper = new ObjectMapper();
+        JsonNode tree = objectMapper.readTree(json);
+        return fromJson(tree);
+    }
+
+    public static Optional<AttributeValue<?>> optionalFromJsonString(String json) {
+        try {
+            return Optional.of(fromJsonString(json));
+        } catch (IOException e) {
+            return Optional.empty();
+        }
+    }
+
+    @VisibleForTesting
+    static AttributeValue<?> fromJson(JsonNode input) {
+        return Optional.of(input)
+                .filter(ObjectNode.class::isInstance)
+                .map(ObjectNode.class::cast)
+                .flatMap(AttributeValue::deserialize)
+                .map(AttributeValue::ofAny)
+                .orElseThrow(() -> new IllegalStateException("unable to deserialize " + input.toString()));
+    }
+
+    public static Optional<?> deserialize(ObjectNode fields) {
+        return Optional.ofNullable(fields.get("serializer"))
```

Comment:
I am afraid that `fields` could be null, then `fields.get...` could throws null pointer exception.

Reply:
I think `fields` being null is a non-sens that should be prevented by our JSON library.

An explicit percondition should be enough IMO.